### ISSUE TITLE
Add the link to the Facility Name on the Appointment Details Page

### ISSUE THREAD

**Issue by ldelacosta** [department-of-veterans-affairs/va.gov-team#89129]
## Description
Add the link to the Facility name on the Appointment Details page. 

Mock-up: 
<img width="338" alt="Screenshot 2024-07-23 at 10 52 01 PM" src="https://github.com/user-attachments/assets/a6339c77-e85a-480c-ba5f-9c96d3830d38">

---

## Requirements 

- [ ] When the user clicks on the Facility link, it will route them to the VA locator that provides additional details of the facility. For example:[ Dayton VA Medical Center](https://www.va.gov/dayton-health-care/locations/dayton-va-medical-center/?_ga=2.34927906.205415193.1721786988-1480407801.1705572091). It does not open to a new page/tab. 

- [ ] Given the user is on the Appointment list for Upcoming appointments
       When the user clicks details for their upcoming VA appointments 
       Then the user will see the link to the Facility name (under Where to attend/Scheduling facility and Need to make changes section, if applicable) 

- [ ] Given the user is on the Appointment list for Upcoming appointments
       When the user clicks details for their upcoming canceled VA appointments 
       Then the user will see the link to the Facility name (under Where to attend/Scheduling facility and Need to make changes section if applicable) 

- [ ] Given the user is on the Appointment list for Pending appointments
       When the user clicks details for their appointment request (VA and Community Care request) 
       Then the user will see the link to the Facility name (under Where section)

- [ ] Given the user is on the Appointment list for Pending appointments
       When the user clicks details for their pending canceled  appointment request (VA and Community Care) 
       Then the user will see the link to the Facility name (under Where to attend/Scheduling facility section)

- [ ] Given the user is on the Appointment list for Past appointments 
       When the user clicks details for their past VA appointments 
       Then the user will see the link to the Facility name (under Where to attend section/Scheduling facility) 

### Design Assets
Not applicable

---
### Development Checklist

- [ ] Task 1
- [ ] Task 2

 ## Developer Reference
Reference this [spike](https://app.zenhub.com/workspaces/appointments-team-603fdef281af6500110a1691/issues/gh/department-of-veterans-affairs/va.gov-team/91474) for more information about the api structure
- [ ] [Swagger file]() if applicable
- [ ] [File page]() if applicable

## Definition of Done
- [ ] All task criteria are met
- [ ] [Technical Documentation](url) is updated

**Comment by Bren22va** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Appointments-Team-603fdef281af6500110a1691/issues/department-of-veterans-affairs/va.gov-team/89129?planning-poker) with Zenhub @cferris32 @jenniemc @JunTaoLuo @ryanshaw @simiadebowale @vbahinwillit

**Comment by Bren22va** [comment]
Need a Spike to determine api info

**Comment by ldelacosta** [comment]
Created the [spike ticket](https://github.com/department-of-veterans-affairs/va.gov-team/issues/89399). cc: @Bren22va @simiadebowale

**Comment by simiadebowale** [comment]
@outerpress - this ticket was blocked but I've unblocked it now that the spike is complete. We can pull it into the next sprint

**Comment by JunTaoLuo** [comment]
@outerpress @melissa-lefevre I see in the description that we should add facility links for the CC appointments too but I'm not sure where the link should be added? There are no "Where to attend/Scheduling facility and Need to make changes section" and the closest section in kind is "Provider". However, in those cases, the provider information are details regarding the CC provider instead of the facility so I think it might be odd to add a link to the facility there?

**Comment by outerpress** [comment]
@JunTaoLuo you're right that we wouldn't add the link to the Provider section for CC appointments so there's nowhere to add it. I'll remove that requirement.

**Comment by JunTaoLuo** [comment]
@outerpress this is ready for validation.

**Comment by outerpress** [comment]
@JunTaoLuo the links themselves work/are accessible. However some are not going to the right page. For example on [this appt](https://staging.va.gov/my-health/appointments/8bd10043008fa90d9c5132deea1c341bc1fcf62af9fe6683f8826ad4f775fafc) in the Judy test user when I click the Cheyenne link it takes me to the [Eastern Colorado Health Care system](https://www.va.gov/eastern-colorado-health-care/). I noticed this with a few other links too. Some are linking to the correct facility, some are not. I can pull up some more examples if it'd be helpful, but if you test a few of Judy's appts you'll see what I mean.

**Comment by JunTaoLuo** [comment]
Ah I missed the part in @jenniemc's comment about hiding this behind a feature toggle for testing in production. Hmmm I wonder if we'll need to wait until she and Stephen are back before we can validate. Adding a feature toggle also might take a bit of time so I doubt we'll be able to fully finish verifying and closing this ticket in the current sprint.

**Comment by JunTaoLuo** [comment]
@simiadebowale had some ideas with how to validate this so I'll assign this to ya for now 😄

**Comment by jenniemc** [comment]
@simiadebowale, I extracted the correct facility website URL from Lighthouse and is documented on [91474](https://github.com/department-of-veterans-affairs/va.gov-team/issues/91474). Can you ask Apothesource Team to replace the test data in staging so UI Team can validate in staging?

**Comment by JRRoof** [comment]
Asked a staff veteran to verify this feature in production.  They had an appointment at Bay Pines and clicking link loads the main BP page.  https://www.va.gov/bay-pines-health-care/locations/cw-bill-young-department-of-veterans-affairs-medical-center/
